PWGHF/vertexingHF/vHFML/AliAnalysisTaskSEDs.cxx

```diff
@@ -1669,8 +1672,10 @@ void AliAnalysisTaskSEDs::CreateCutVarsAndEffSparses()
   Double_t maxMass = massDs + 0.5 * nInvMassBins * fMassBinSize;
 
   Int_t nSparseAxes = knVarForSparse;
-  if (!fApplyML)
-    nSparseAxes--;
+  if (!fApplyML && !fMultiClass)
+    nSparseAxes -= 3;
+  if (fApplyML && !fMultiClass)
+    nSparseAxes -= 2;
   if (fApplyML && fUseMinimalVarForSparse)
     nSparseAxes = knVarForSparseMLMinimal;
```

Comment:
Hi @stefanopolitano, now the logic starts to become a bit cumbersome. You can rewrite everything as:
```
Int_t nSparseAxes = knVarForSparse;
if (!fApplyML) {
    nSparseAxes -= 3;
} else {
    if(fUseMinimalVarForSparse)
        nSparseAxes = knVarForSparseMLMinimal;
    if (!fMultiClass)
        nSparseAxes -= 2;
}